rpc/kate-rpc/src/lib.rs

```diff
@@ -418,6 +418,32 @@ where
 			.ok_or_else(|| internal_err!("Missing block hash {:?}", at))?
 			.block;
 
+		// data index starts from 0
+		let mut data_idx: i32 = -1;
```

Comment:
I think it should be easier if you `filter` first and then use `nth` function. Something like:
```Rust
if let Some((data_idx, _ex)) = block.extrinsics().iter().enumerate()
   .filter( |(_data_idx,ex)| self.client.runtime_api().<filter_function>(&ex))
   .nth( transaction_index) {
   
      // use `data_idx` here
   }

Reply:
moved logic after initial filtering when calling `calls_proof`